lib/core-integration/src/Test/Integration/Framework/DSL.hs

```diff
@@ -865,6 +873,31 @@ eventuallyUsingDelay delay io = do
         threadDelay delay
         trial
 
+-- similar as @eventually@ but returns IO a
+eventually'
+    :: IO a
+    -> IO a
+eventually' = eventuallyUsingDelay' (500 * ms)
+  where
+    ms = 1000
+
+-- similar as @eventuallyUsingDelay@ but returns IO a
+eventuallyUsingDelay'
+    :: Int
+    -> IO a
+    -> IO a
+eventuallyUsingDelay' delay io = do
```

Comment:
@piotr-iohk A more idiomatic Haskell approach for this would be to have the default function `eventuallyUsingDelay` work from `IO a` -> `IO a` and define a `eventuallyUsingDelay_` working with `IO a -> IO ()` and simply implemented as the composition of `void` and `eventuallyUsingDelay`. Very much like:

- [forM](https://hackage.haskell.org/package/base-4.12.0.0/docs/Control-Monad.html#v:forM) vs [forM_](https://hackage.haskell.org/package/base-4.12.0.0/docs/Control-Monad.html#v:forM_) 
- [bracket](https://hackage.haskell.org/package/base-4.12.0.0/docs/Control-Exception.html#v:bracket) vs [bracket_](https://hackage.haskell.org/package/base-4.12.0.0/docs/Control-Exception.html#v:bracket_)
- [modifyMVar](https://hackage.haskell.org/package/base-4.12.0.0/docs/Control-Concurrent-MVar.html#v:modifyMVar) vs [modifyMVar_](https://hackage.haskell.org/package/base-4.12.0.0/docs/Control-Concurrent-MVar.html#v:modifyMVar_) 

etc..